server/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java

```diff
@@ -328,14 +335,14 @@ public boolean clearData(String nodeName) {
      */
     public void testIndexDeletionWhenNodeRejoins() throws Exception {
         final String indexName = "test-index-del-on-node-rejoin-idx";
-        final int numNodes = 2;
+        // We need at least 3 nodes to make sure, that once one node is stopped, remaining nodes can elect a new master
```

Comment:
I think the proper fix here is to add the down-scale / up-scale logic that's in InternalTestCluster#stopNodesAndClients to restartNode as well, just as we do with the autoManageMinMasterNodes scaling. The test should then not need any adaptation

Reply:
https://github.com/elastic/elasticsearch/pull/36367/commits/6e5bbd4cc235aeadbf7a72d2f4642635c98690ec